src/robot/utils/robotio.py

```diff
@@ -41,3 +43,17 @@ def binary_file_writer(path=None):
     getvalue = f.getvalue
     f.getvalue = lambda encoding='UTF-8': getvalue().decode(encoding)
     return f
+
+
+def _ensure_destination_directory_exists(path):
+    if not os.path.exists(path):
+        base, ext = path.rsplit(os.sep, 1)
```

Comment:
This line actually has a bug. If the path is given as just a file name like `syslog.txt`, the above line fails with `ValueError: need more than 1 value to unpack`. Better to use `os.path.dirname(path)` and then also check `if dirname and not os.path.exists(dirname)`. I'll fix this myself while doing some other cleanup to creating destination directories.